x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/support/ApiKeyBoolQueryBuilderTests.java

```diff
@@ -60,17 +67,120 @@
+            assertQueryFields(queryFields, qb, null);
+            assertCommonFilterQueries(apiKeyQb, null);
+            List<QueryBuilder> mustQueries = apiKeyQb.must();
+            assertThat(mustQueries, hasSize(1));
+            assertThat(mustQueries.get(0), equalTo(qb));
+            assertThat(apiKeyQb.should(), emptyIterable());
+            assertThat(apiKeyQb.mustNot(), emptyIterable());
+        }
+        {
+            Authentication authentication = AuthenticationTests.randomAuthentication(null, null);
+            QueryBuilder qb = randomSimpleQuery("name");
+            List<String> queryFields = new ArrayList<>();
+            ApiKeyBoolQueryBuilder apiKeyQb = ApiKeyBoolQueryBuilder.build(qb, queryFields::add, authentication);
+            assertQueryFields(queryFields, qb, authentication);
+            assertCommonFilterQueries(apiKeyQb, authentication);
+            List<QueryBuilder> mustQueries = apiKeyQb.must();
+            assertThat(mustQueries, hasSize(1));
+            assertThat(mustQueries.get(0), equalTo(qb));
+            assertThat(apiKeyQb.should(), emptyIterable());
+            assertThat(apiKeyQb.mustNot(), emptyIterable());
+        }
+        {
+            String apiKeyId = randomUUID();
+            Authentication authentication = AuthenticationTests.randomApiKeyAuthentication(AuthenticationTests.randomUser(), apiKeyId);
+            QueryBuilder qb = randomSimpleQuery("name");
+            List<String> queryFields = new ArrayList<>();
+            ApiKeyBoolQueryBuilder apiKeyQb = ApiKeyBoolQueryBuilder.build(qb, queryFields::add, authentication);
+            assertQueryFields(queryFields, qb, authentication);
+            assertCommonFilterQueries(apiKeyQb, authentication);
+            List<QueryBuilder> mustQueries = apiKeyQb.must();
+            assertThat(mustQueries, hasSize(1));
+            assertThat(mustQueries.get(0), equalTo(qb));
+            assertThat(apiKeyQb.should(), emptyIterable());
+            assertThat(apiKeyQb.mustNot(), emptyIterable());
+        }
+    }
+
+    public void testMatchQueryBuilderPropertiesArePreservedFor() {
+        // the match query has many properties, that all must be preserved after limiting for API Key docs only
+        Authentication authentication = randomFrom(
+            AuthenticationTests.randomApiKeyAuthentication(AuthenticationTests.randomUser(), randomUUID()),
+            AuthenticationTests.randomAuthentication(null, null),
+            null
+        );
+        String fieldName = randomFrom(
+            "username",
+            "realm_name",
+            "name",
+            "type",
+            "creation",
+            "expiration",
+            "invalidated",
+            "invalidation",
+            "metadata",
+            "metadata.what.ever"
+        );
+        MatchQueryBuilder matchQueryBuilder = QueryBuilders.matchQuery(fieldName, new Object());
+        if (randomBoolean()) {
```

Comment:
If `boost` is missing in the query builder it would be nice with a test for it here.

Reply:
I've added the test here: [06252a9](https://github.com/elastic/elasticsearch/pull/104594/commits/06252a99888bf33cac71f6403b0aea3107b32009)